frontend/src/components/users/ShowUsers.jsx

```diff
@@ -0,0 +1,191 @@
+import React from 'react';
+import { useHistory } from 'react-router-dom';
+// eslint-disable-next-line import/no-extraneous-dependencies
+import moment from 'moment';
+import PropTypes from 'prop-types';
+
+import Typography from '@material-ui/core/Typography';
+import { makeStyles } from '@material-ui/core';
+import Link from '@material-ui/core/Link';
+
+import usersAPI from '../../helpers/api/users';
+
+const useStyles = makeStyles((theme) => ({
+  root: {
+    height: 'calc(100% - 64px)',
+  },
+
+  inventory: {
+    fontSize: '0.875rem',
+    marginBottom: theme.spacing(2),
+  },
+
+  /* flex */
+  flexWrapper: {
+    padding: '0 16px',
+  },
+
+  flex: {
+    display: '-ms-flexbox',
+    // eslint-disable-next-line no-dupe-keys
+    display: 'flex',
+    position: 'relative',
+  },
+
+  /* Title row */
+  flexTitle: {
+    padding: '8px 0',
+    borderBottom: '2px solid #c0c0c0',
+  },
+
+  /* Data Row */
+  flexRow: {
+    padding: '8px 0',
+    borderBottom: '1px solid #c0c0c0',
+    '&:hover': {
+      boxShadow: '0 0 3px 0 rgba(0, 0, 0, 0.8)',
+    },
+  },
+
+  /* Column definiions */
+  flexCol1: {
+    flex: '1 1 0',
+    marginRight: '8px',
+    paddingLeft: '8px',
+    minWidth: '0',
+  },
+
+  flexCol2: {
+    flex: '2 1 0',
+    marginRight: '8px',
+    paddingLeft: '8px',
+    minWidth: '0',
+  },
+
+  flexCol3: {
+    flex: '3 1 0',
+    marginRight: '8px',
+    paddingLeft: '8px',
+    minWidth: '0',
+  },
+
+  flexCol4: {
+    flex: '4 1 0',
+    marginRight: '8px',
+    paddingLeft: '8px',
+    minWidth: '0',
+  },
+
+  flexColAuto: {
+    width: 'auto',
+    marginRight: '8px',
+    paddingLeft: '8px',
+    minWidth: '0',
+  },
+
+  /* Use to scale and hide columns in the title row */
+  flexColAutoHidden: {
+    width: 'auto',
+    marginRight: '8px',
+    paddingLeft: '8px',
+    minWidth: '0',
+    visibility: 'hidden',
+  },
+
+  show: {
+    display: 'block',
+  },
+
+  hide: {
+    display: 'none',
+  },
+
+  pointer: {
+    cursor: 'pointer',
+  },
+}));
+
+// eslint-disable-next-line no-unused-vars, object-curly-newline
+const ShowUsers = ({ users, setIsLoading, setAlertProps, permissionsList, currentLetter }) => {
+  const classes = useStyles();
+  const keyRetired = Object.keys(permissionsList).find((key) => permissionsList[key] === 'retired');
+  const history = useHistory();
+
+  const toggleRetire = async (user, key, val) => {
+    const formData = {
+      user_id: user.id,
+      permission_id: key,
+      value: val,
+    };
+
+    const response = await usersAPI.permissionUpdate(formData);
+    if (!response) return;
+
+    // success
+    if (val === 'on') {
+      // eslint-disable-next-line no-param-reassign, operator-assignment, prefer-template
+      user.permission_ids = user.permission_ids + '6.';
```

Comment:
Avoid use of permission values in constants  since they are configurable in the backend

Reply:
This is done and pushed. However, something happened in your commit

014eba0d86da2c8fb6477972edc5195916262bcc "Merge branch 'v3-refactor' into merge_samples_objects"

that blew up eslint.  The code now does not compile in the sample_types area. I will leave it to you and Mariko to clean up since I don't know what else may or may not have changed...(If I just fix the eslint stuff I may be missing other logic that was in the branch before the merge)

Here is the full list:

src/components/sampleTypes/fields/SampleTypeFieldForm.spec.js
  Line 3:1:     Expected indentation of 2 spaces but found 1  indent
  Line 3:10:    'fireEvent' is defined but never used         no-unused-vars
  Line 3:38:    'wait' is defined but never used              no-unused-vars
  Line 3:51:    Trailing spaces not allowed                   no-trailing-spaces
  Line 18:24:   A space is required after '{'                 object-curly-spacing
  Line 18:41:   A space is required before '}'                object-curly-spacing
  Line 18:44:   A space is required after '{'                 object-curly-spacing
  Line 18:61:   A space is required before '}'                object-curly-spacing
  Line 35:7:    Missing semicolon                             semi
  Line 42:42:   A space is required after '{'                 object-curly-spacing
  Line 42:55:   A space is required before '}'                object-curly-spacing
  Line 105:1:   Expected indentation of 4 spaces but found 3  indent
  Line 106:1:   Expected indentation of 6 spaces but found 5  indent
  Line 108:1:   Expected indentation of 6 spaces but found 5  indent
  Line 109:1:   Expected indentation of 4 spaces but found 3  indent
  Line 111:1:   Expected indentation of 4 spaces but found 3  indent
  Line 112:1:   Expected indentation of 6 spaces but found 5  indent
  Line 113:1:   Expected indentation of 6 spaces but found 5  indent
  Line 115:1:   Expected indentation of 6 spaces but found 5  indent
  Line 117:1:   Expected indentation of 6 spaces but found 5  indent
  Line 118:1:   Expected indentation of 4 spaces but found 3  indent
  Line 120:1:   Expected indentation of 4 spaces but found 3  indent
  Line 121:1:   Expected indentation of 6 spaces but found 5  indent
  Line 122:1:   Expected indentation of 6 spaces but found 5  indent
  Line 124:1:   Expected indentation of 6 spaces but found 5  indent
  Line 126:1:   Expected indentation of 6 spaces but found 5  indent
  Line 127:1:   Expected indentation of 4 spaces but found 3  indent
  Line 160:10:  Block must not be padded by blank lines       padded-blocks
  Line 162:1:   More than 1 blank line not allowed            no-multiple-empty-lines
  Line 290:8:   Missing semicolon                             semi
  Line 302:42:  A space is required after '{'                 object-curly-spacing
  Line 302:58:  A space is required before '}'                object-curly-spacing
  Line 351:41:  A space is required after '{'                 object-curly-spacing
  Line 351:62:  A space is required before '}'                object-curly-spacing
  Line 368:9:   Missing semicolon                             semi
  Line 370:5:   Missing semicolon                             semi

src/components/shared/Buttons.spec.js
  Line 2:8:    A space is required after '{'            object-curly-spacing
  Line 2:34:   A space is required before '}'           object-curly-spacing
  Line 2:65:   Missing semicolon                        semi
  Line 5:8:    A space is required after '{'            object-curly-spacing
  Line 5:35:   A space is required before '}'           object-curly-spacing
  Line 19:48:  A space is required after '{'            object-curly-spacing
  Line 19:66:  A space is required before '}'           object-curly-spacing
  Line 39:30:  A space is required after ','            comma-spacing
  Line 39:40:  Block must not be padded by blank lines  padded-blocks

src/components/shared/DropdownMenu.spec.js
  Line 1:8:   'React' is defined but never used         no-unused-vars
  Line 2:10:  'render' is defined but never used        no-unused-vars
  Line 2:18:  'screen' is defined but never used        no-unused-vars
  Line 3:8:   'DropdownMenu' is defined but never used  no-unused-vars

src/components/shared/LoadingSpinner.spec.js
  Line 11:1:  Expected indentation of 4 spaces but found 6  indent
  Line 12:1:  Expected indentation of 4 spaces but found 6  indent
  Line 17:1:  Expected indentation of 4 spaces but found 6  indent
  Line 19:1:  Expected indentation of 4 spaces but found 6  indent
  Line 20:1:  Expected indentation of 4 spaces but found 6  indent
  Line 21:1:  Expected indentation of 4 spaces but found 6  indent

src/components/shared/Modal.spec.js
  Line 3:1:   Expected indentation of 2 spaces but found 1  indent
  Line 3:37:  Trailing spaces not allowed                   no-trailing-spaces
  Line 15:1:  More than 1 blank line not allowed            no-multiple-empty-lines
  Line 28:1:  Expected indentation of 4 spaces but found 2  indent
  Line 38:8:  Block must not be padded by blank lines       padded-blocks

src/components/shared/Select.spec.js
  Line 18:4:   Missing semicolon               semi
  Line 35:11:  A space is required after '{'   object-curly-spacing
  Line 35:21:  A space is required before '}'  object-curly-spacing
  Line 54:3:   Missing semicolon               semi

src/components/shared/TextInput.spec.js
  Line 7:37:  Missing semicolon             semi
  Line 18:6:  Strings must use singlequote  quotes